Sword Trail Revamp

shun-one-1 opened the pull request:
HDR Assets: [sword_trail_revamp.zip](https://github.com/user-attachments/files/23012228/sword_trail_revamp.zip)

PR that redesigns all sword trails in the game to be thinner and more uniform. It also adds new sword trail effects for various special moves across the cast.

**Aegis**

Pyra/Mythra
[$] New sword trail

Mythra
Lightning Buster
[$] New sword trail

Chroma Dust
[$] New sword trail

<video src=https://github.com/user-attachments/assets/c6d047f6-1411-4466-87b9-01eec5d3cc69 width="320" height="240" controls></video>

https://github.com/user-attachments/assets/ea4964bc-2291-4fee-b471-fa1cc46635c2

**Byleth**

[$] New sword trail
<video src=https://github.com/user-attachments/assets/4636a5cf-e4aa-4315-91c2-18a093947420></video>


**Chrom**

[$] New sword trail
<video src=https://github.com/user-attachments/assets/d13809c1-1d2f-41a8-ae06-ef2515187e98></video

[$] Special trail cleaned up
Old:
<img width="1920" height="1080" alt="483836780-1a937e4e-aba0-449d-8a7e-44aa781e41b7" src="https://github.com/user-attachments/assets/8de873a5-1684-4b48-be3e-ac83b327573c" />

New:
<img width="1920" height="1080" alt="Super Smash Bros  Ultimate - akeru_2025-10-13_13-49-08" src="https://github.com/user-attachments/assets/273d7bb6-0715-41f6-98ca-51a42e67406d" />

**Cloud**

[$] New sword trail
<video src=https://github.com/user-attachments/assets/66df1b6f-b4c7-49aa-8183-c0786b914207></video>

Cross Slash
[$] New sword trail
<video src=https://github.com/user-attachments/assets/e04b5297-a247-41dc-9da3-386c629d8341></video>

Climhazard 
[$] New sword trail
<video src=https://github.com/user-attachments/assets/f67962d3-3d19-4e3e-8d48-ce8dfcb01433></video>

Finishing Touch 
[$] New sword trail
<video src=https://github.com/user-attachments/assets/c45945d5-da6f-40f4-838f-dd6d8babee2b></video>

**Corrin**

[$] New sword trail
<video src=https://github.com/user-attachments/assets/8559f72e-2b63-40e7-976b-5f85a6ceb413></video>

**Dark Pit**

[$] New sword trail
<video src=https://github.com/user-attachments/assets/6b35bffe-b468-404a-b791-206092584de8></video>

**Ganondorf**

[$] New sword trail
<video src=https://github.com/user-attachments/assets/895312b2-7617-4da4-96c2-7d9d987d63c7></video>

**Hero**

[$] New sword trail
<video src=https://github.com/user-attachments/assets/79fbeb07-584f-4091-b03a-d692d55d4bd9></video>

F-Smash Psyche Up
[$] Uses Hatchet Man trail

D-Smash 1 Psyche Up
[$] Uses Flame Slash trail

D-Smash 2 Psyche Up
[$] Uses Kakrackle Slash trail
<video src=https://github.com/user-attachments/assets/b66cd4e7-b628-44a7-ae35-d01a9c4e1e92></video>

Kazap
[$] New sword trail
<video src=https://github.com/user-attachments/assets/3f05a941-cea9-4ab6-8e16-4f173c09190b></video>

Flame Slash
[$] New sword trail
<video src=https://github.com/user-attachments/assets/742ce3f0-edb6-45dc-b203-43cc841cace4></video>

Kakrackle Slash
[$] New sword trail
<video src=https://github.com/user-attachments/assets/e62753a4-96f0-47c7-9fa6-125db5c6d4ac></video>

Hatchet Man
[$] New sword trail
<video src=https://github.com/user-attachments/assets/7c662a14-1c75-4761-842b-ef791a3f72c0></video>


**Ike**

[$] new sword trail
<video src=https://github.com/user-attachments/assets/367f8863-0cdb-43e0-8508-67deb8f04301></video>

F Smash, Down Smash, Up Smash, D-air
[$] new sword trail
<video src=https://github.com/user-attachments/assets/759fce96-c040-43d5-949b-fd2577b7543b></video>

Quickdraw
[$] New sword trail
<video src=https://github.com/user-attachments/assets/fa43907a-ffc6-4b69-92bd-fe8c97442c74></video>

Aether
[$] New sword trail
<video src=https://github.com/user-attachments/assets/f5da75b3-a888-44b6-a788-a34f62fb47fe></video>

**Link**
[$] new sword trail

Spin Attack
[$] New sword trail
<video src=https://github.com/user-attachments/assets/ee4f94bf-8d77-44f5-9e4b-aed815a4a6c2></video>

**Lucina**
[$] new sword trail

Armor Breaker
[$] changed glow to green
[$] changed sword glow to green

Dolphin Slash
[$] New sword trail
[$] Changed sword glow to green
<video src=https://github.com/user-attachments/assets/f9f54938-b843-4bfe-b0d9-0597f2e8ca3a></video>

**Marth**
[$] new sword trail

Dolphin Slash
[$] new sword trail
<video src=https://github.com/user-attachments/assets/a25e7d8f-a131-4757-8260-b9d600b93b9f></video>

Counter
[$] new sword trail
<video src=https://github.com/user-attachments/assets/4a96a18d-a1f5-4c29-9baa-b3fb59881a09></video>

**Meta Knight**
[$] New sword trail

Shuttle Loop
[$] New sword trail

Dimensional Cape
[$] New sword trail
<video src=https://github.com/user-attachments/assets/2e906673-9c2d-4698-913d-152c5e9a9ee1></video>

**Pit**
[$] New sword trail
<video src=https://github.com/user-attachments/assets/118ef9e3-1ace-4fea-a165-05cec81a78cc></video>

**Robin**

[$] new sword trail
<video src=https://github.com/user-attachments/assets/2ef68acd-1557-4c28-97b6-ed5b4ea9d5e3></video>

All Levin Sword Attacks
[$] new sword trail
<video src=https://github.com/user-attachments/assets/4d75117f-55b1-40ae-b22b-72eefd481828></video>

**Roy** 

[$] new sword trails
<video src=https://github.com/user-attachments/assets/e958ff9f-8dcd-471d-b0c7-8c3de7a6f0fa></video>

Flare Blade
[$] New sword trail

Blazer
[$] new sword trail
<video src=https://github.com/user-attachments/assets/feab273b-a8df-4902-b29d-8b8f2a6ad16c></video>

Counter
[$] uses flame trail from Flare Blade
<video src=https://github.com/user-attachments/assets/d830416b-e03c-494d-9cc9-98b36f4ca95c></video>

**Sephiroth**
[$] new sword trail
<video src=https://github.com/user-attachments/assets/86ff4a75-d320-49be-8576-7614cb4a5b39></video>

**Shulk**
[$] new sword trail

Heavy Rush
[$] new sword trail

Air Slash
[$] new sword trail
<video src=https://github.com/user-attachments/assets/c81426eb-778a-4832-a0ff-7eb41bd2b545></video>

**Sora**
[$] new sword trail

Aerial Spin
[$] new sword trail
<video src=https://github.com/user-attachments/assets/1d64b334-7fcf-4aca-b7d9-ae35ea7268f0></video>

**Toon Link**
[$] new sword trail

Spin Attack
[$] new sword trail
<video src=https://github.com/user-attachments/assets/750a9031-807c-425c-9cea-b46401a1c228></video>

**Young Link**
[$] New sword trail

Spin Attack
[$] New sword trail
<video src=https://github.com/user-attachments/assets/cedb24ba-56d9-4554-9556-76e30ddf3099></video>

**Zelda**

Phantom Sword 1
[$] New sword trail
<video src=https://github.com/user-attachments/assets/a9ec828a-ebf5-465a-9a87-6388c7af35fa></video>

Phantom Sword 2
[$] New sword trail
<video src=https://github.com/user-attachments/assets/83290d2b-f15b-4e10-ba83-9d485e482d6b></video>

Phantom Sword 3
[$] New sword trail
<video src=https://github.com/user-attachments/assets/0f56255b-1b45-4e4f-85fd-8f7287ccfcd7></video>

shun-one-1 committed:
Sword Trail Revamp

PR that redesigns all sword trails in the game to be thinner and more uniform. It also adds new sword trail effects for various special moves across the cast.

shun-one-1 committed:
Sword Trail Revamp

SuddyN commented:
please upload the required assets to this PR in a zip folder

shun-one-1 commented:
> please upload the required assets to this PR in a zip folder

Should be uploaded now

shun-one-1 committed:
Fixes Pit and Dark Pit rapid jab

Fixes Rapid jab for Pit and Dark Pit not ending.

shun-one-1 committed:
Metaknight

Adds Metaknight effect changes

shun-one-1 committed:
Zelda Phantom Edits

adds trail edits to Zelda's phantom.

shun-one-1 committed:
Additional Polish

properly installs effect scripts for Link's Spin Attack and Marth and Mythra's tilts. Cleans up Ike's code spacing.

Updates assets for Link's Spin Attack with an effect transplant and polishs Marth's sword trail

shun-one-1 committed:
Update specials.rs

shun-one-1 committed:
Ike polish

shun-one-1 committed:
Update specials.rs

shun-one-1 committed:
Chrom polish

Updates special trail and cleans some sword glow calls.

shun-one-1 committed:
Cloud Side-B Fix

Fixes kanji vanishing when facing right on Cross Slash

SuddyN commented on fighters/brave/src/acmd/specials/special_lw.rs:
```diff
@@ -171,6 +171,30 @@ unsafe extern "C" fn effect_speciallw17(agent: &mut L2CAgentBase) {
     }
 }
 
+unsafe extern "C" fn effect_specialairlw17(agent: &mut L2CAgentBase) {
+    frame(agent.lua_state_agent, 3.0);
+    if is_excute(agent) {
+        EFFECT_FOLLOW_WORK(agent, *FIGHTER_BRAVE_INSTANCE_WORK_ID_INT_EFFECT_KIND_LIGHTNING_SWORD_FLARE, Hash40::new("sword1"), 0, 0, 0, 0, 0, -90, 1, true);
+        EFFECT_FOLLOW_NO_STOP(agent, Hash40::new("brave_fire_sword"), Hash40::new("sword1"), 0, 0, 0, 0, 0, -90, 1.45, true);
+    }
+    frame(agent.lua_state_agent, 5.0);
+    if is_excute(agent) {
+        AFTER_IMAGE4_ON_arg29(agent, Hash40::new("tex_brave_flamesword"), Hash40::new("tex_brave_sword2"), 7, Hash40::new("sword1"), 1.5, 0, 0, Hash40::new("sword1"), 14.4, 0, 0, true, Hash40::new("null"), Hash40::new("sword1"), 0, 0, 0, 0, 0, -90, 1, 0, *EFFECT_AXIS_X, 0, *TRAIL_BLEND_ALPHA, 101, *TRAIL_CULL_NONE, 1.4, 0.1);
+    }
+    frame(agent.lua_state_agent, 11.0);
+    if is_excute(agent) {
+        AFTER_IMAGE_OFF(agent, 4);
+    }
+    frame(agent.lua_state_agent, 13.0);
+    if is_excute(agent) {
+        EFFECT_OFF_KIND(agent, Hash40::new("brave_fire_sword"), false, true);
+    }
+    frame(agent.lua_state_agent, 15.0);
+    if is_excute(agent) {
+        EFFECT_OFF_KIND_WORK(agent, *FIGHTER_BRAVE_INSTANCE_WORK_ID_INT_EFFECT_KIND_LIGHTNING_SWORD_FLARE, false, true);
+    }
+}
```
this function is not installed

SuddyN commented on fighters/brave/src/acmd/specials/mod.rs:
```diff
@@ -226,6 +226,130 @@ unsafe extern "C" fn game_specials32(agent: &mut L2CAgentBase) {
+frame(lua_state, 21.0);
+if is_excute(agent) {
+    EFFECT_FOLLOW(agent, Hash40::new("brave_lightning3_swordspark"), Hash40::new("sword1"), 0, 0, 0, 0, 0, -90, 1, true);
+}
+frame(lua_state, 22.0);
+if is_excute(agent) {
+    AFTER_IMAGE4_ON_arg29(agent, Hash40::new("tex_brave_zapsword"), Hash40::new("tex_brave_sword2"), 16, Hash40::new("sword1"), 3, 0, 0, Hash40::new("sword1"), 14, 0, 0, true, Hash40::new("null"), Hash40::new("sword1"), 0, 0, 0, 0, 0, -90, 1, 0, *EFFECT_AXIS_X, 0, *TRAIL_BLEND_ALPHA, 101, *TRAIL_CULL_NONE, 1.4, 0.1);
+}
+frame(lua_state, 26.0);
+if is_excute(agent) {
+    LANDING_EFFECT(agent, Hash40::new("sys_whirlwind_r"), Hash40::new("top"), 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, false);
+    LAST_EFFECT_SET_RATE(agent, 0.8);
+}
+frame(lua_state, 28.0);
+if PostureModule::lr(agent.module_accessor) < 0.0 {
+    if is_excute(agent) {
+        EFFECT_FOLLOW(agent, Hash40::new("brave_lightning3_rolllightning"), Hash40::new("top"), 0, 10, 0, 0, 0, -5, 1, true);
+        EFFECT_OFF_KIND(agent, Hash40::new("brave_lightning3_swordspark"), false, true);
+    }
+}
+frame(lua_state, 30.0);
+if is_excute(agent) {
+    EFFECT_FOLLOW(agent, Hash40::new("brave_lightning3_swordspark"), Hash40::new("sword1"), 0, 0, 0, 0, 0, -90, 1, true);
+}
+else {
+if is_excute(agent) {
+    EFFECT_FOLLOW(agent, Hash40::new("brave_lightning3_rolllightning"), Hash40::new("top"), 0, 10, 0, 0, 0, 4, 1, true);
+}
+}
+frame(lua_state, 34.0);
+if is_excute(agent) {
+EFFECT_OFF_KIND(agent, Hash40::new("brave_lightning3_swordspark"), false, true);
+}
+frame(lua_state, 40.0);
+if is_excute(agent) {
+AFTER_IMAGE_OFF(agent, 10);
+EFFECT_OFF_KIND_WORK(agent, *FIGHTER_BRAVE_INSTANCE_WORK_ID_INT_EFFECT_KIND_LIGHTNING_SWORD_FLARE, false, true);
+EFFECT_OFF_KIND(agent, Hash40::new("brave_lightning3_sword"), false, true);
+}
+}
```
please fix formatting

SuddyN commented on fighters/brave/src/acmd/specials/mod.rs:
```diff
@@ -226,6 +226,130 @@ unsafe extern "C" fn game_specials32(agent: &mut L2CAgentBase) {
+    wait(lua_state, 2.0);
+    if is_excute(agent) {
+        COL_NORMAL(agent);
+    }
+}
+frame(lua_state, 21.0);
+if is_excute(agent) {
+    EFFECT_FOLLOW(agent, Hash40::new("brave_lightning3_swordspark"), Hash40::new("sword1"), 0, 0, 0, 0, 0, -90, 1, true);
+}
+frame(lua_state, 22.0);
+if is_excute(agent) {
+    AFTER_IMAGE4_ON_arg29(agent, Hash40::new("tex_brave_zapsword"), Hash40::new("tex_brave_sword2"), 16, Hash40::new("sword1"), 3, 0, 0, Hash40::new("sword1"), 14, 0, 0, true, Hash40::new("null"), Hash40::new("sword1"), 0, 0, 0, 0, 0, -90, 1, 0, *EFFECT_AXIS_X, 0, *TRAIL_BLEND_ALPHA, 101, *TRAIL_CULL_NONE, 1.4, 0.1);
+}
+frame(lua_state, 28.0);
+if PostureModule::lr(agent.module_accessor) < 0.0 {
+    if is_excute(agent) {
+        EFFECT_FOLLOW(agent, Hash40::new("brave_lightning3_rolllightning"), Hash40::new("top"), 0, 10, 0, 0, 0, -5, 1, true);
+        EFFECT_OFF_KIND(agent, Hash40::new("brave_lightning3_swordspark"), false, true);
+    }
+}
+frame(lua_state, 30.0);
+if is_excute(agent) {
+    EFFECT_FOLLOW(agent, Hash40::new("brave_lightning3_swordspark"), Hash40::new("sword1"), 0, 0, 0, 0, 0, -90, 1, true);
+}
+else {
+if is_excute(agent) {
+    EFFECT_FOLLOW(agent, Hash40::new("brave_lightning3_rolllightning"), Hash40::new("top"), 0, 10, 0, 0, 0, 4, 1, true);
+}
+}
+frame(lua_state, 34.0);
+if is_excute(agent) {
+EFFECT_OFF_KIND(agent, Hash40::new("brave_lightning3_swordspark"), false, true);
+}
+frame(lua_state, 40.0);
+if is_excute(agent) {
+AFTER_IMAGE_OFF(agent, 10);
+EFFECT_OFF_KIND_WORK(agent, *FIGHTER_BRAVE_INSTANCE_WORK_ID_INT_EFFECT_KIND_LIGHTNING_SWORD_FLARE, false, true);
+EFFECT_OFF_KIND(agent, Hash40::new("brave_lightning3_sword"), false, true);
+}
+}
```
not installed. also fix formatting

SuddyN commented on fighters/brave/src/acmd/specials/special_lw.rs:
```diff
@@ -236,6 +260,30 @@ unsafe extern "C" fn effect_speciallw18(agent: &mut L2CAgentBase) {
     }
 }
 
+unsafe extern "C" fn effect_specialairlw18(agent: &mut L2CAgentBase) {
+    frame(agent.lua_state_agent, 4.0);
+    if is_excute(agent) {
+        EFFECT_FOLLOW_WORK(agent, *FIGHTER_BRAVE_INSTANCE_WORK_ID_INT_EFFECT_KIND_SWORD_FLARE, Hash40::new("sword1"), 0, 0, 0, 0, 0, -90, 1, true);
+    }
+    frame(agent.lua_state_agent, 6.0);
+    if is_excute(agent) {
+        AFTER_IMAGE4_ON_arg29(agent, Hash40::new("tex_brave_icesword"), Hash40::new("tex_brave_sword2"), 8, Hash40::new("sword1"), 1.5, 0, 0, Hash40::new("sword1"), 14.4, 0, 0, true, Hash40::new("null"), Hash40::new("sword1"), 0, 0, 0, 0, 0, -90, 1, 0, *EFFECT_AXIS_X, 0, *TRAIL_BLEND_ALPHA, 101, *TRAIL_CULL_NONE, 1.4, 0.1);
+    }
+    frame(agent.lua_state_agent, 7.0);
+    if is_excute(agent) {
+        EFFECT_FOLLOW_NO_STOP(agent, Hash40::new("brave_ice_sword"), Hash40::new("sword1"), 0, 0, 0, 0, 0, -90, 1.5, true);
+    }
+    frame(agent.lua_state_agent, 11.0);
+    if is_excute(agent) {
+        AFTER_IMAGE_OFF(agent, 6);
+        EFFECT_OFF_KIND_WORK(agent, *FIGHTER_BRAVE_INSTANCE_WORK_ID_INT_EFFECT_KIND_SWORD_FLARE, false, true);
+    }
+    frame(agent.lua_state_agent, 12.0);
+    if is_excute(agent) {
+        EFFECT_OFF_KIND(agent, Hash40::new("brave_ice_sword"), false, true);
+    }
+}
```
not installed

SuddyN commented on fighters/elight/src/acmd/aerials.rs:
```diff
@@ -55,6 +55,59 @@ unsafe extern "C" fn game_attackairn(agent: &mut L2CAgentBase) {
+        LAST_EFFECT_SET_OFFSET_TO_CAMERA_FLAT(agent, -0.3);
+    }
+    frame(lua_state, 6.0);
+    if is_excute(agent) {
+        EFFECT_FOLLOW(agent, Hash40::new("elight_sword_light_rdm"), Hash40::new("sword1"), 0, 0, 0, 0, 0, 0, 1, true);
+        EffectModule::enable_sync_init_pos_last(boma);
+    }
+    frame(lua_state, 14.0);
+    if is_excute(agent) {
+        EFFECT_FOLLOW(agent, Hash40::new("elight_sword_light"), Hash40::new("sword1"), 0, 0, 0, 0, 0, 0, 1, true);
+        EffectModule::enable_sync_init_pos_last(boma);
+    }
+    frame(lua_state, 15.0);
+    if is_excute(agent) {
+        EFFECT_FOLLOW(agent, Hash40::new("elight_airn_slash"), Hash40::new("top"), 0, 0, 0, 0, 0, 0, 1, true);
+    }
+    frame(lua_state, 17.0);
+    if is_excute(agent) {
+        EFFECT_DETACH_KIND(agent, Hash40::new("elight_airn_slash"), -1);
+    }
+    frame(lua_state, 18.0);
+    if is_excute(agent) {
+        EFFECT_OFF_KIND(agent, Hash40::new("elight_sword_light_rdm"), false, true);
+    }
+    frame(lua_state, 19.0);
+    if is_excute(agent) {
+        EFFECT_OFF_KIND(agent, Hash40::new("elight_sword_light"), false, true);
+    }
+    frame(lua_state, 20.0);
+    if is_excute(agent) {
+        AFTER_IMAGE_OFF(agent, 3);
+    }
+    frame(lua_state, 22.0);
+    if is_excute(agent) {
+        EFFECT_OFF_KIND(agent, Hash40::new("elight_sword_beam_m"), true, true);
+        EFFECT_FOLLOW(agent, Hash40::new("elight_sword_close_m"), Hash40::new("sword1"), 0, 0, 0, 0, 90, 0, 1, true);
+        LAST_EFFECT_SET_OFFSET_TO_CAMERA_FLAT(agent, -0.3);
+    }
+}
+
```
not installed

SuddyN commented on fighters/elight/src/acmd/aerials.rs:
```diff
@@ -95,6 +148,38 @@ unsafe extern "C" fn game_attackairf(agent: &mut L2CAgentBase) {
     }
 }
 
+unsafe extern "C" fn effect_attackairf(agent: &mut L2CAgentBase) {
+    let lua_state = agent.lua_state_agent;
+    let boma = agent.boma();
+    frame(lua_state, 5.0);
+    if is_excute(agent) {
+        EFFECT_FOLLOW(agent, Hash40::new("elight_sword_open"), Hash40::new("sword1"), 0, 0, 0, 0, 90, 0, 1, true);
+        LAST_EFFECT_SET_OFFSET_TO_CAMERA_FLAT(agent, -0.3);
+    }
+    frame(lua_state, 7.0);
+    if is_excute(agent) {
+        AFTER_IMAGE4_ON_arg29(agent, Hash40::new("tex_elight_sword_hdr"), Hash40::new("tex_elight_sword2"), 10, Hash40::new("sword1"), 0, 0, -0.08, Hash40::new("sword1"), 14.5, 0, -0.08, true, Hash40::new("null"), Hash40::new("sword1"), 0, 0, 0, 0, 0, 0, 1, 0, *EFFECT_AXIS_X, 0, *TRAIL_BLEND_ALPHA, 101, *TRAIL_CULL_NONE, 1.4, 0.1);
+        EFFECT_OFF_KIND(agent, Hash40::new("elight_sword_open"), true, true);
+        EFFECT_FOLLOW(agent, Hash40::new("elight_sword_beam_s"), Hash40::new("sword1"), 0, 0, 0, 0, 90, 0, 1, true);
+        LAST_EFFECT_SET_OFFSET_TO_CAMERA_FLAT(agent, -0.3);
+        EFFECT_FOLLOW(agent, Hash40::new("elight_sword_light"), Hash40::new("sword1"), 0, 0, 0, 0, 0, 0, 1, true);
+        EffectModule::enable_sync_init_pos_last(boma);
+    }
+    frame(lua_state, 8.0);
+    if is_excute(agent) {
+        EFFECT_FOLLOW(agent, Hash40::new("elight_airf_slash"), Hash40::new("top"), 0, 0, 0, 0, 0, 0, 1, true);
+    }
+    frame(lua_state, 11.0);
+    if is_excute(agent) {
+        EFFECT_DETACH_KIND(agent, Hash40::new("elight_airf_slash"), -1);
+        EFFECT_OFF_KIND(agent, Hash40::new("elight_sword_light"), false, true);
+        AFTER_IMAGE_OFF(agent, 2);
+        EFFECT_OFF_KIND(agent, Hash40::new("elight_sword_beam_s"), true, true);
+        EFFECT_FOLLOW(agent, Hash40::new("elight_sword_close_s"), Hash40::new("sword1"), 0, 0, 0, 0, 90, 0, 1, true);
+        LAST_EFFECT_SET_OFFSET_TO_CAMERA_FLAT(agent, -0.3);
+    }
+}
```
not installed

SuddyN commented on fighters/elight/src/acmd/aerials.rs:
```diff
@@ -140,6 +225,36 @@ unsafe extern "C" fn game_attackairb(agent: &mut L2CAgentBase) {
     }
 }
 
+unsafe extern "C" fn effect_attackairb(agent: &mut L2CAgentBase) {
+    let lua_state = agent.lua_state_agent;
+    let boma = agent.boma();
+    frame(lua_state, 10.0);
+    if is_excute(agent) {
+        EFFECT_FOLLOW(agent, Hash40::new("elight_sword_open"), Hash40::new("sword1"), 0, 0, 0, 0, 90, 0, 1, true);
+        LAST_EFFECT_SET_OFFSET_TO_CAMERA_FLAT(agent, -0.3);
+    }
+    frame(lua_state, 12.0);
+    if is_excute(agent) {
+        EFFECT_OFF_KIND(agent, Hash40::new("elight_sword_open"), true, true);
+        EFFECT_FOLLOW(agent, Hash40::new("elight_sword_beam_m"), Hash40::new("sword1"), 0, 0, 0, 0, 90, 0, 1, true);
+        LAST_EFFECT_SET_OFFSET_TO_CAMERA_FLAT(agent, -0.3);
+    }
+    frame(lua_state, 13.0);
+    if is_excute(agent) {
+        AFTER_IMAGE4_ON_arg29(agent, Hash40::new("tex_elight_sword_hdr"), Hash40::new("tex_elight_sword2"), 6, Hash40::new("sword1"), 0, 0, -0.08, Hash40::new("sword1"), 15.5, 0, -0.08, true, Hash40::new("null"), Hash40::new("sword1"), 0, 0, 0, 0, 0, 0, 1, 0, *EFFECT_AXIS_X, 0, *TRAIL_BLEND_ALPHA, 101, *TRAIL_CULL_NONE, 1.4, 0.1);
+        EFFECT_FOLLOW(agent, Hash40::new("elight_sword_light_few"), Hash40::new("sword1"), 0, 0, 0, 0, 0, 0, 1, true);
+        EffectModule::enable_sync_init_pos_last(boma);
+    }
+    frame(lua_state, 16.0);
+    if is_excute(agent) {
+        EFFECT_OFF_KIND(agent, Hash40::new("elight_sword_light_few"), false, true);
+        AFTER_IMAGE_OFF(agent, 3);
+        EFFECT_OFF_KIND(agent, Hash40::new("elight_sword_beam_m"), true, true);
+        EFFECT_FOLLOW(agent, Hash40::new("elight_sword_close_m"), Hash40::new("sword1"), 0, 0, 0, 0, 90, 0, 1, true);
+        LAST_EFFECT_SET_OFFSET_TO_CAMERA_FLAT(agent, -0.3);
+    }
+}
+
```
not installed

SuddyN commented on fighters/elight/src/acmd/aerials.rs:
```diff
@@ -198,6 +313,39 @@ unsafe extern "C" fn game_attackairhi(agent: &mut L2CAgentBase) {
     }
 }
 
+unsafe extern "C" fn effect_attackairhi(agent: &mut L2CAgentBase) {
+    let lua_state = agent.lua_state_agent;
+    let boma = agent.boma();
+    frame(lua_state, 8.0);
+    if is_excute(agent) {
+        EFFECT_FOLLOW(agent, Hash40::new("elight_sword_open"), Hash40::new("sword1"), 0, 0, 0, 0, 90, 0, 1, true);
+        LAST_EFFECT_SET_OFFSET_TO_CAMERA_FLAT(agent, -0.3);
+    }
+    frame(lua_state, 10.0);
+    if is_excute(agent) {
+        AFTER_IMAGE4_ON_arg29(agent, Hash40::new("tex_elight_sword_hdr"), Hash40::new("tex_elight_sword2"), 6, Hash40::new("sword1"), 0, 0, -0.08, Hash40::new("sword1"), 15.5, 0, -0.08, true, Hash40::new("null"), Hash40::new("sword1"), 0, 0, 0, 0, 0, 0, 1, 0, *EFFECT_AXIS_X, 0, *TRAIL_BLEND_ALPHA, 101, *TRAIL_CULL_NONE, 1.4, 0.1);
+    }
+    frame(lua_state, 11.0);
+    if is_excute(agent) {
+        EFFECT_OFF_KIND(agent, Hash40::new("elight_sword_open"), true, true);
+        EFFECT_FOLLOW(agent, Hash40::new("elight_sword_beam_m"), Hash40::new("sword1"), 0, 0, 0, 0, 90, 0, 1, true);
+        LAST_EFFECT_SET_OFFSET_TO_CAMERA_FLAT(agent, -0.3);
+        EFFECT_FOLLOW(agent, Hash40::new("elight_sword_light_few"), Hash40::new("sword1"), 0, 0, 0, 0, 0, 0, 1, true);
+        EffectModule::enable_sync_init_pos_last(boma);
+    }
+    frame(lua_state, 16.0);
+    if is_excute(agent) {
+        EFFECT_OFF_KIND(agent, Hash40::new("elight_sword_light_few"), false, true);
+        AFTER_IMAGE_OFF(agent, 3);
+    }
+    frame(lua_state, 17.0);
+    if is_excute(agent) {
+        EFFECT_OFF_KIND(agent, Hash40::new("elight_sword_beam_m"), true, true);
+        EFFECT_FOLLOW(agent, Hash40::new("elight_sword_close_m"), Hash40::new("sword1"), 0, 0, 0, 0, 90, 0, 1, true);
+        LAST_EFFECT_SET_OFFSET_TO_CAMERA_FLAT(agent, -0.3);
+    }
+}
+
```
not installed

SuddyN commented on fighters/elight/src/acmd/aerials.rs:
```diff
@@ -250,6 +398,35 @@ unsafe extern "C" fn game_attackairlw(agent: &mut L2CAgentBase) {
     }
 }
 
+unsafe extern "C" fn effect_attackairlw(agent: &mut L2CAgentBase) {
+    let lua_state = agent.lua_state_agent;
+    let boma = agent.boma();
+    frame(lua_state, 12.0);
+    if is_excute(agent) {
+        EFFECT_FOLLOW(agent, Hash40::new("elight_sword_open"), Hash40::new("sword1"), 0, 0, 0, 0, 90, 0, 1, true);
+        LAST_EFFECT_SET_OFFSET_TO_CAMERA_FLAT(agent, -0.3);
+    }
+    frame(lua_state, 14.0);
+    if is_excute(agent) {
+        AFTER_IMAGE4_ON_arg29(agent, Hash40::new("tex_elight_sword_hdr"), Hash40::new("tex_elight_sword2"), 6, Hash40::new("sword1"), 0, 0, -0.08, Hash40::new("sword1"), 15.5, 0, -0.08, true, Hash40::new("null"), Hash40::new("sword1"), 0, 0, 0, 0, 0, 0, 1, 0, *EFFECT_AXIS_X, 0, *TRAIL_BLEND_ALPHA, 101, *TRAIL_CULL_NONE, 1.4, 0.1);
+        EFFECT_OFF_KIND(agent, Hash40::new("elight_sword_open"), true, true);
+        EFFECT_FOLLOW(agent, Hash40::new("elight_sword_beam_m"), Hash40::new("sword1"), 0, 0, 0, 0, 90, 0, 1, true);
+        LAST_EFFECT_SET_OFFSET_TO_CAMERA_FLAT(agent, -0.3);
+        EFFECT_FOLLOW(agent, Hash40::new("elight_sword_light_few"), Hash40::new("sword1"), 0, 0, 0, 0, 0, 0, 1, true);
+    }
+    frame(lua_state, 18.0);
+    if is_excute(agent) {
+        EFFECT_OFF_KIND(agent, Hash40::new("elight_sword_light_few"), false, true);
+    }
+    frame(lua_state, 19.0);
+    if is_excute(agent) {
+        AFTER_IMAGE_OFF(agent, 3);
+        EFFECT_OFF_KIND(agent, Hash40::new("elight_sword_beam_m"), true, true);
+        EFFECT_FOLLOW(agent, Hash40::new("elight_sword_close_m"), Hash40::new("sword1"), 0, 0, 0, 0, 90, 0, 1, true);
+        LAST_EFFECT_SET_OFFSET_TO_CAMERA_FLAT(agent, -0.3);
+    }
+}
+
```
not installed

SuddyN commented on fighters/metaknight/src/acmd/tilts.rs:
```diff
@@ -74,6 +74,22 @@ unsafe extern "C" fn game_attacks3s2(agent: &mut L2CAgentBase) {
     }
 }
 
+unsafe extern "C" fn effect_attacks3s2(agent: &mut L2CAgentBase) {
+    if is_excute(agent) {
+        EFFECT_FOLLOW(agent, Hash40::new("metaknight_sword"), Hash40::new("haver"), 0, 0, 0, 0, 0, 0, 1, true);
+        FOOT_EFFECT(agent, Hash40::new("sys_turn_smoke"), Hash40::new("top"), 0, 0, 0, 0, 0, 0, 1.2, 2, 0, 0, 0, 0, 0, false);
+    }
+    frame(agent.lua_state_agent, 2.0);
+    if is_excute(agent) {
+        EFFECT_FOLLOW(agent, Hash40::new("metaknight_s2_trace_hdr"), Hash40::new("top"), -0.0, 0, 0, 0, 0, 0, 1, true);
+        EffectModule::set_disable_render_offset_last(agent.module_accessor);
+    }
+    frame(agent.lua_state_agent, 4.0);
+    if is_excute(agent) {
+        EFFECT_OFF_KIND(agent, Hash40::new("metaknight_sword"), false, false);
+    }
+}
+
```
not installed

SuddyN commented on fighters/metaknight/src/acmd/tilts.rs:
```diff
@@ -88,6 +104,26 @@ unsafe extern "C" fn game_attacks3s3(agent: &mut L2CAgentBase) {
     }
 }
 
+unsafe extern "C" fn effect_attacks3s3(agent: &mut L2CAgentBase) {
+    if is_excute(agent) {
+        EFFECT_FOLLOW(agent, Hash40::new("metaknight_sword"), Hash40::new("haver"), 0, 0, 0, 0, 0, 0, 1, true);
+        FOOT_EFFECT(agent, Hash40::new("sys_atk_smoke"), Hash40::new("top"), -6, 0, 0, 0, 0, 0, 0.6, 0, 0, 0, 0, 0, 0, false);
+    }
+    frame(agent.lua_state_agent, 2.0);
+    if is_excute(agent) {
+        EFFECT_FOLLOW(agent, Hash40::new("metaknight_s3_trace_hdr"), Hash40::new("top"), -0.0, 0, 0, 0, 0, 0, 1, true);
+        EffectModule::set_disable_render_offset_last(agent.module_accessor);
+    }
+    frame(agent.lua_state_agent, 4.0);
+    if is_excute(agent) {
+        EFFECT(agent, Hash40::new("sys_smash_flash"), Hash40::new("haver"), 0, 14, 0, 0, 0, 0, 0.8, 0, 0, 0, 0, 0, 0, true);
+    }
+    frame(agent.lua_state_agent, 6.0);
+    if is_excute(agent) {
+        EFFECT_OFF_KIND(agent, Hash40::new("metaknight_sword"), false, false);
+    }
+}
+
```
not installed

shun-one-1 committed:
Requested Fixes

Addresses Hero, Mythra, and Metaknight not installed issues and formatting issues.
